src/python/turicreate/toolkits/sound_classifier/sound_classifier.py

```diff
@@ -105,14 +106,47 @@ def reset(self):
         # instance itself.
         pass
 
+class _NumPyDataIterator(_DataIterator):
+    def __init__(self, data, label=None, batch_size=1, shuffle=False):
+
+        # Always pass a tuple, so that the impl's built-in iterator returns a
+        # tuple.
+        self.data_slices = (data, label) if label is not None else (data,)
+        self.batch_size = batch_size
+        self.batch_idx = 0
+        self.num_batches = int(_ceil(len(data) / float(self.batch_size)))
+
+    def __iter__(self):
```

Comment:
This looks like an implementation of `__next__` for an iterator, not an implementation of `__iter__` itself...

Shouldn't `__iter__` just return an iterator? Or self if we're building the iterator logic into the main class?

Reply:
Yes, great catch! Thank you